### DISCUSSION TITLE
State Dict

### DISCUSSION THREAD

**Discussion by AmeerArsala** [Toni-SM/skrl#118] [Q&A] [answered]
Hi there, 

I've been trying to use skrl's JAX version and got an error that said the models didn't have `state_dict` (I was trying to use PPO) and I couldn't find anything about it anywhere on the documentation.

I eventually found out how to deal with this by looking at one of the examples, where it is essentially just looping over the model dictionary and initializing the state dict. Is there some reason this isn't done automatically in the PPO constructor?

**Comment by Toni-SM** [comment, marked as answer]
Hi @AmeerArsala 

The initialization of the model `state_dict` is not performed inside the agent to ensure modularity of the components with as little interference as possible. This makes the library more flexible for customized modifications and makes the code visible to the user. 

Thanks to your question, I have updated the documentation to include model `state_dict` initialization note and snippets
